### PR TITLE
Fixes #338811 - Remove deprecated things

### PR DESCRIPTION
**Pull request by ekohl**
Replaces https://github.com/theforeman/smart-proxy/pull/804.

### CODE REVIEW

File: lib/proxy/plugins.rb [hunk 1]
```diff
@@ -50,28 +50,4 @@ def select
...
-  def find_plugin(plugin_name)
-    p = loaded.find { |plugin| plugin[:name] == plugin_name.to_sym }
-    return p[:class] if p
-  end
-
-  def find_provider(provider_name)
-    provider = loaded.find { |p| p[:name] == provider_name.to_sym }
-    raise ::Proxy::PluginProviderNotFound, "Provider '#{provider_name}' could not be found" if provider.nil? || !provider[:class].ancestors.include?(::Proxy::Provider)
-    provider[:class]
-  end
```
Comment (ekohl) [inline lib/proxy/plugins.rb:76]: I am debating about this: we can modify them in 3.4 to log deprecation warnings when used.

File: lib/proxy/provider_factory.rb [hunk 2]
```diff
@@ -1,9 +0,0 @@
-class ::Proxy::ProviderFactory
-  class << self
-    def get_provider(provider_name, opts = {})
```
Comment (ekohl) [inline lib/proxy/provider_factory.rb:3]: Should also be modified to log a deprecation warning in 3.4.

File: lib/proxy/plugins.rb (hunk 1 above)
Comment (ekohl) [reply lib/proxy/plugins.rb:76]: https://github.com/theforeman/smart-proxy/pull/841

File: lib/proxy/provider_factory.rb (hunk 2 above)
Comment (ekohl) [reply lib/proxy/provider_factory.rb:3]: https://github.com/theforeman/smart-proxy/pull/841

File: lib/proxy/http_download.rb [hunk 3]
```diff
@@ -5,11 +5,8 @@ class HttpDownload < Proxy::Util::CommandTask
     include Util
     DEFAULT_CONNECT_TIMEOUT = 10
 
-    def initialize(src, dst, read_timeout = nil, connect_timeout = nil, dns_timeout = nil, verify_server_cert = false)
+    def initialize(src, dst, connect_timeout: DEFAULT_CONNECT_TIMEOUT, verify_server_cert: false)
       @dst = dst
-      logger.warn('Deprecated: HttpDownload read_timeout is deprecated and will be removed in 4.0') if read_timeout
-      logger.warn('Deprecated: HttpDownload dns_timeout is deprecated and will be removed in 4.0') if dns_timeout
-      connect_timeout ||= DEFAULT_CONNECT_TIMEOUT
```
Comment (evgeni) [inline lib/proxy/http_download.rb:12]: technically, we're not releasing 4.0 right now, so we can't remove that here?

Comment (ekohl) [reply lib/proxy/http_download.rb:12]: Hmm, good question. Should we change it to "removed in 3.5" in https://github.com/theforeman/smart-proxy/pull/841 instead?

Comment (evgeni) [reply lib/proxy/http_download.rb:12]: I'd be cool with that.

Comment (ekohl) [reply lib/proxy/http_download.rb:12]: Done and merged.

**Pull request merged**